### ISSUE TITLE
Several features unavailable from admin panel

### ISSUE THREAD

**Issue by matthias4217** [YunoHost-Apps/dotclear2_ynh#40]
### Describe the bug

I have installed DotClear 2 from the Webadmin (I have also tried with `yunohost app install dotclear2` with the same result), on the `/` path with an unused domain (I have tried with a `/test` to see if the problem came from the path, but with the same results). The application was installed, and I was able to log to the admin. I can write articles, but it seems that several functionalities are missing. I cannot create pages, categories etc... Below is a screenshot of how the admin panel looks, which I believe is not how it should be. I have tried to manually access pages such as `https://domain.tld/test/admin/pages.php` or `https://domain.tld/test/admin/categories.php`, but it doesn't seem to be working (I don't know if these particular urls should work though).

![Capture d’écran du 2022-01-05 19-19-36](https://user-images.githubusercontent.com/17904729/148268483-7b61f9b1-2ed6-4a90-9335-7b96a7f888b0.png)

My guess is that I may not have the right permissions, as `/var/www/dotclear2/admin/users.php` or `/var/www/dotclear2/admin/categories.php` do exist. But I don't know why it would be that way, and how I could check.

### Context

- Hardware: VPS
- YunoHost version: 4.3.5
- I have access to my server: *Through SSH | through the webadmin*
- Are you in a special context or did you perform some particular tweaking on your YunoHost instance?: *I don't think so.*
- Using, or trying to install package version/branch: Main version

### Steps to reproduce

Install the app, either from the webadmin or the CLI (`sudo yunohost app install dotclear2`). Login as an admin. The dashboard is emptied from several features.

### Expected behavior

Having every feature available

Thanks for having a look. I might have more configuration to do, but I don't think I've seen any instruction about this in the README or somewhere else. Sorry to bother you if I've just made a mistake somewhere.

**Comment by kay0u** [comment]
Hello !

Well, there is a limitation I imposed (for simplicity reasons) when I wrote the ldap login of dotclear, but we can easily modify it if necessary.

The user you give during the installation is the administrator of dotclear, and each other users have the authorization "usage":
https://github.com/YunoHost-Apps/dotclear2_ynh/blob/ccca78c3c40ab688fb0216eccf7aace5c0b313c5/conf/class.auth.ldap.php#L61-L72 (note the `array('usage'=>true)` which is the line that gave the corresponding rights). We can add more permissions if needed here.

In short: I can write a config panel to manage the permissions of YNH users.

If you don't want to wait that, you can login with the account you gave during the installation which is "super" admin of dotclear

**Comment by matthias4217** [comment]
Thank you for your answer !

However, I was logging in with my admin account `alice`, which I have set up during installation of DotClear to be the admin. So I went to `domain.tld/dotclear-path/admin`, typed `alice` and `alice-yunohost-password` to login. I can log in, but only write (and not publish) blog posts, so it's clear it hasn't got DotClear admin permissions.

I disconnected, and logged on yunohost with my `bob` testing account. However, going to `domain.tld/dotclear-path/admin` wouldn't work. This is normal behaviour as in Yunohost Groups and permissions management, only `alice` has the permission for *DotClear2 (Admin)*.

From this, I understand `alice` has been selected as the admin account during the installation, and that on the Yunohost side, everything is okay. However, on the DotClear side, there is an issue somewhere.

**Comment by kay0u** [comment]
Hmm that's strange!

If you want, you can install phpmyadmin, go to `dotclear2->dc_user` table, and change the value of `user_super` to `1` for the row `alice`.

I am writing a config_panel to manage dotclear permissions to simplify/automate this task.

**Comment by matthias4217** [comment]
Thank you for the `phpMyAdmin` advice ! I've found out the issue.

Two users have been created for DotClear2 : `alice`, and `Alice`. I was logging in with `Alice` (`user_super` is `NULL` in phMyAdmin), which hadn't got any admin permission. Logging with `alice` (`user_super` is `1`), I was a superadmin. Both have the same password.

On the Yunohost user panel, I can see that my username is `alice`, but I generally log in on Yunohost with `Alice`, but logging in with `alice`works just as well.

I don't know exactly where the issue is, but at least you've solved my issue ! Thank you very much for your help !

**Comment by kay0u** [comment]
The issue is probably somewhere here: https://github.com/YunoHost-Apps/dotclear2_ynh/blob/ccca78c3c40ab688fb0216eccf7aace5c0b313c5/conf/class.auth.ldap.php#L23-L28

In YunoHost do you have one user `alice` and one `Alice`? Or only one user `alice` with firstname `Alice`?

You're welcome, I'll try to reproduce this issue on my side

**Comment by matthias4217** [comment]
I've got `alice` as a username, but a totally different full name : `John` `Doe`. The full name is right in the DotClear database for both `alice` and `Alice`. However, I have written `Alice` in my password manager, so I believe I have used `Alice` when installing Yunohost, and it got converted afterwards, though I'm not 100% sure.

**Comment by kay0u** [comment]
I'm not sure to understand, do you have a user `Alice` in YunoHost?

**Comment by matthias4217** [comment]
No, I haven't got a user with username `Alice`. I only have one `alice`.

**Comment by kay0u** [comment]
I've found the issue, https://github.com/YunoHost-Apps/dotclear2_ynh/pull/41/commits/f2b7228a4a1ffd4d86563c250b2cf394a2592b5d should fix it :+1:

Many thanks for the report!
